### PR TITLE
SONAR-7168 don't query Components of CETasks 1 by 1 in TaskFormatter

### PR DESCRIPTION
**Pull request by sns-seb**
This PR also includes some optimisations for `DatabaseUtils.executeLargeInputs`

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java [hunk 1]
```diff
@@ -142,25 +142,50 @@ private static void buildComponent(WsCe.Task.Builder builder, @Nullable Componen
...
+      .filter(notNull())
+      .toSet();
+  }
+
+  private static Set<String> ceActivityDtoToComponentUuids(Iterable<CeActivityDto> dtos) {
+    return from(dtos)
+      .transform(new Function<CeActivityDto, String>() {
+        @Override
+        @Nullable
+        public String apply(@Nonnull CeActivityDto input) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java]: MAJOR The "input" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

File: server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java [hunk 2]
```diff
@@ -175,4 +200,32 @@ Long computeExecutionTimeMs(CeQueueDto dto) {
...
+  private final class CeActivityDtoToTask implements Function<CeActivityDto, WsCe.Task> {
+    private final ComponentDtoCache cache;
+
+    public CeActivityDtoToTask(ComponentDtoCache cache) {
+      this.cache = cache;
+    }
+
+    @Override
+    @Nonnull
+    public WsCe.Task apply(@Nonnull CeActivityDto input) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java]: MAJOR The "input" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)
MAJOR The return value nullability of this method is different in the superclass, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

File: server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java [hunk 3]
```diff
@@ -142,25 +142,50 @@ private static void buildComponent(WsCe.Task.Builder builder, @Nullable Componen
...
+          return input.getComponentUuid();
+        }
+      })
+      .filter(notNull())
+      .toSet();
+  }
+
+  private static Set<String> ceActivityDtoToComponentUuids(Iterable<CeActivityDto> dtos) {
+    return from(dtos)
+      .transform(new Function<CeActivityDto, String>() {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java]: MAJOR Make this a named "static" inner class. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)

File: server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java [hunk 4]
```diff
@@ -142,25 +142,50 @@ private static void buildComponent(WsCe.Task.Builder builder, @Nullable Componen
     }
   }
 
-  private class ComponentCache {
-    private final DbSession dbSession;
-    private final Map<String, ComponentDto> componentsByUuid = new HashMap<>();
+  private static Set<String> ceQueueDtoToComponentUuids(Iterable<CeQueueDto> dtos) {
+    return from(dtos)
+      .transform(new Function<CeQueueDto, String>() {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java]: MAJOR Make this a named "static" inner class. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2694)

File: server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java [hunk 5]
```diff
@@ -175,4 +200,32 @@ Long computeExecutionTimeMs(CeQueueDto dto) {
...
+  private final class CeQueueDtoToTask implements Function<CeQueueDto, WsCe.Task> {
+    private final ComponentDtoCache cache;
+
+    public CeQueueDtoToTask(ComponentDtoCache cache) {
+      this.cache = cache;
+    }
+
+    @Override
+    @Nonnull
+    public WsCe.Task apply(@Nonnull CeQueueDto input) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java]: MAJOR The "input" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)
MAJOR The return value nullability of this method is different in the superclass, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

File: server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java [hunk 6]
```diff
@@ -142,25 +142,50 @@ private static void buildComponent(WsCe.Task.Builder builder, @Nullable Componen
...
 
-  private class ComponentCache {
-    private final DbSession dbSession;
-    private final Map<String, ComponentDto> componentsByUuid = new HashMap<>();
+  private static Set<String> ceQueueDtoToComponentUuids(Iterable<CeQueueDto> dtos) {
+    return from(dtos)
+      .transform(new Function<CeQueueDto, String>() {
+        @Override
+        @Nullable
+        public String apply(@Nonnull CeQueueDto input) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/ws/TaskFormatter.java]: MAJOR The "input" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

**Comment by SonarTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 9 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by julienlancelot** [comment]
LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 9 issues:
* MAJOR 9 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**